server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleDetailResp.java

```diff
@@ -163,8 +175,31 @@ public ScheduleDetailResp map(@NonNull ScheduleEntity entity) {
             if (datasource != null) {
                 resp.setDatasource(new InnerConnection(datasource));
             }
-
-            resp.setJobParameters(entity.getJobParametersJson());
+            switch (entity.getJobType()) {
+                case DATA_ARCHIVE: {
+                    DataArchiveParameters parameters =
+                            JsonUtils.fromJson(entity.getJobParametersJson(), DataArchiveParameters.class);
+                    RateLimitConfiguration rateLimitConfig = getDLMRateLimitConfigurationById.apply(entity.getId());
+                    if (rateLimitConfig != null) {
+                        parameters.setRateLimit(rateLimitConfig);
+                    }
+                    resp.setJobParameters(JsonUtils.toJson(parameters));
+                    break;
+                }
+                case DATA_DELETE: {
+                    DataDeleteParameters parameters =
+                            JsonUtils.fromJson(entity.getJobParametersJson(), DataDeleteParameters.class);
+                    RateLimitConfiguration rateLimitConfig = getDLMRateLimitConfigurationById.apply(entity.getId());
+                    if (rateLimitConfig != null) {
+                        parameters.setRateLimit(rateLimitConfig);
+                    }
+                    resp.setJobParameters(JsonUtils.toJson(parameters));
```

Comment:
look like many duplicate code DATA_DELETE with DATA_ARCHIVE in switch

Reply:
Simplified code implementation